tests/messages/consumed/bioscan_pool_xp_to_traction/test_request.py

```diff
@@ -0,0 +1,80 @@
+from typing import Callable
+from unittest.mock import patch
+
+import pytest
+from tests.data.example_bioscan_pool_xp_to_traction_messages import VALID_REQUEST_PAYLOAD
+from tol_lab_share.constants.input_bioscan_pool_xp_to_traction_message import (
+    REQUEST_COST_CODE,
+    REQUEST_GENOME_SIZE,
+    REQUEST_LIBRARY_TYPE,
+    REQUEST_STUDY_UUID,
+)
+from tol_lab_share.messages.consumed.bioscan_pool_xp_to_traction.request import Request
+
+
+@pytest.fixture
+def subject():
+    return Request(VALID_REQUEST_PAYLOAD.copy(), "path", "parent_path")
+
+
+class TestRequest:
+    def test_init_passes_arguments_to_record(self):
+        path = "path"
+        parent_path = "parent_path"
+
+        with patch("tol_lab_share.messages.consumed.bioscan_pool_xp_to_traction.request.Record.__init__") as super_init:
+            Request(VALID_REQUEST_PAYLOAD, path, parent_path)
+
+        super_init.assert_called_once_with(VALID_REQUEST_PAYLOAD, path, parent_path)
+
+    def test_cost_code_returns_correct_message_field(self, subject):
+        with patch(
+            "tol_lab_share.messages.consumed.bioscan_pool_xp_to_traction.request.Request._make_field"
+        ) as make_field:
+            field = subject.cost_code
+
+        make_field.assert_called_once_with(REQUEST_COST_CODE)
+        assert field == make_field.return_value
+
+    def test_cost_code_returns_correct_value(self, subject):
+        assert subject.cost_code.value == VALID_REQUEST_PAYLOAD[REQUEST_COST_CODE]
+
+    def test_genome_size_returns_correct_message_field(self, subject):
+        with patch(
+            "tol_lab_share.messages.consumed.bioscan_pool_xp_to_traction.request.Request._make_field"
+        ) as make_field:
+            field = subject.genome_size
+
+        make_field.assert_called_once_with(REQUEST_GENOME_SIZE)
+        assert field == make_field.return_value
+
+    def test_genome_size_returns_correct_value(self, subject):
+        assert subject.genome_size.value == VALID_REQUEST_PAYLOAD[REQUEST_GENOME_SIZE]
+
+    def test_genome_size_returns_none_when_not_present(self, subject):
+        subject._payload.pop(REQUEST_GENOME_SIZE)
+        assert subject.genome_size.value is None
```

Comment:
I suppose by the time the request arrives to tol-lab-share, the integrity of these values would have been verified, and there's no need to check whether they are valid or not? (like, can the `genomeSize` have values without the unit `bp`?)

Reply:
It's not our job to determine what the sender should have included or what the recipient should accept.  We are middleware, concerned only with routing the information.  Any validation beyond that would go in the validator class.  This layer is only concerned with processing what we received.  There are restrictions on types, but we have to have faith that Avro is dealing with that for us.  Hence the addition of the `fastavro` validation test in the `lab-share-lib`.